examples/save_config_demo/data/main.js

```diff
@@ -18,8 +18,16 @@ $( document ).ready(function() {
          url: '/settings',
          success: function(data) {
            $.each(data, function(key, value, data) {
-             var input = "input:text[name='" + key + "']";
-             $(input).val(value)
+             var input = document.getElementsByName(key);
+             if (input.length > 0) {
+               var dataType = input[0].getAttribute("data-type");
+               if (dataType == "boolean") {
+                 $(input[0]).prop("checked", value);
+                 return
+               }
+
+               $(input[0]).val(value);
```

Comment:
I'd get rid of the nested return in favor of a `else` so the logic is a bit clearer at first glance .. 

```
if (dataType == "boolean") {
 $(input[0]).prop("checked", value);
} else {
 $(input[0]).val(value);
}

```

Reply:
I would tend to disagree. Else is widely considered a code smell, preferring a return in the if.

See:
https://wiki.c2.com/?ElseConsideredSmelly

Reply:
I think we are reading that page differently ;) -- as it told me 5 lines are OK .. your call however

if we are taking that page verbatim .. 

we don't fail a complicated condition .. 
```
When the condition() is the least bit complicated, the else is twice as complicated, because the reader has to invert the condition().
```
The ifblock is not unreasonable
```
When the ifBlock() contains a little more than reasonable number of lines it is easy to forget what the condition() was.
```

we dont have nesting
```
Nested if-if-else, if-else-if (ArrowAntiPattern), etc. are likely to confuse the reader even more. Nested if's without any else's are not so bad.
```

And I feel this lands more on this, where that page says its AOK

```
So, there's nothing really wrong with "else" at all: It's just that "else" happened to be seen where other code smells were present.
"Else" still adds the complexity of inverting the condition. This is specific to "else". It makes smell one and two worse and smell three much worse. cf. also next discussion point.
Both first and second smells are trivial to fix, if you just have the Then and Else clauses consist solely of a call to a subroutine.
 if (condition()) {
   do_true_handling();
 } else {
   do_false_handling();
```

Reply:
although that does say sub routine ... your call ... I just feel randomly placed returns lead to confusion as you can easily overlook .. but then I also personally hate breaks ;)

Reply:
Yeah, it is simple enough to keep the else, the issue for me with "else" in simple cases is maintainability. In general when expansion is needed, people tend to just toss an "else if" in. Not having the else means you end up with your early exit cases indented and your main "happy path" unindented. In this case it probably is very much personal preference though.

Reply:
Cool, either way .. it looks good

Reply:
Thanks, appreciate the feedback 😄